server/schedule/placement/fit.go

```diff
@@ -180,12 +182,21 @@ func newFitWorker(stores []*core.StoreInfo, region *core.RegionInfo, rules []*Ru
 		return si > sj || (si == sj && peers[i].GetId() < peers[j].GetId())
 	})
 
+	matchRules := make(map[int]map[int]bool, len(rules))
+	for id, rule := range rules {
+		matchPeers := make(map[int]bool, 0)
+		if checkRule(rule, stores) {
+			matchRules[id] = make(map[int]bool)
+		}
+		matchRules[id] = matchPeers
```

Comment:
It seems `matchRules[id]` is always equal to `matchPeers`, so what is the purpose of the if condition here?